corehq/apps/reports/util.py

```diff
@@ -453,12 +453,17 @@ def add_tableau_user(domain, username):
     these details to the Tableau instance.
     '''
     session = TableauAPISession.create_session_for_domain(domain)
+    if TableauUser.objects.filter(
```

Comment:
This is a good place to use `get_or_create` to avoid race conditions where the object gets created by a different process after the `exists` but before the `create`. Highly unlikely at the scale this feature is used, just a good habit.